clippy_lints/src/disallowed_methods.rs

```diff
@@ -49,7 +49,7 @@ declare_clippy_lint! {
     /// ```
     #[clippy::version = "1.49.0"]
     pub DISALLOWED_METHODS,
-    nursery,
+    style,
```

Comment:
I'm not sure if style is the right group (I felt like it is "user-defined code styles"), but I couldn't find another group that looked good. (restriction that #7841 originally proposed is allowed by default.)